api/v1/controllers/samples.js

```diff
@@ -128,6 +144,7 @@ module.exports = {
    */
   putSample(req, res, next) {
     utils.noReadOnlyFieldsInReq(req, helper);
+    rejectIfNameInBody(req.body);
     doPut(req, res, next, helper);
```

Comment:
bulkUpsert and upsert also use noReadOnlyFieldsInReq. Adding the name field would reject upserts and bulk upserts